### DISCUSSION TITLE
instalation Micropython with Thonny on ESP-WROOM-02D not work

### DISCUSSION THREAD

**Discussion by DanSkareda** [micropython/micropython#13503] [ESP8266]
Hello. I have an ESP-WROOM-02D board (contains ESP8266ex). I am connected to it by a UART programmer. I can upload projects from the Arduino IDE and they work. So I started Thonny and wanted to use it to upload the Micropython firmware. The process went through, but the board in the shell window started broadcasting endless "red nonsense". Everything is in the attachment. I am doing something wrong? The same procedure worked for me in case of ESP32C3 (with micropython firmware for ESP32C3), the board responded correctly. I try again and again but every time I flash micropython firmware for ESP8266 to ESP8266Ex it does the same "red nonsense".
![1](https://github.com/micropython/micropython/assets/81017132/3edec838-d711-4567-be43-fcf095becdc6)
![2](https://github.com/micropython/micropython/assets/81017132/8d40d2b2-4e11-49be-87b0-07ac29a6b34a)
![3](https://github.com/micropython/micropython/assets/81017132/8d5980af-24e1-47b2-90e2-a1f7fd6bf2f9)
![4](https://github.com/micropython/micropython/assets/81017132/724af60c-3ddf-4768-8f97-dd3be5d898c8)

**Comment by robert-hh** [comment]
Did you try the different `flash mode` options when uploading the firmware? DOUT is a common alternative.

**Reply by DanSkareda** [reply]
Thanks for advice. The correct choice was DIO.
![20](https://github.com/micropython/micropython/assets/81017132/ee418422-c275-43d1-84e0-3ef2b1180138)
![21](https://github.com/micropython/micropython/assets/81017132/e588c8bb-cabc-4e14-86be-8a6d707e387b)

**Comment by DanSkareda** [comment]
I tried all possible flash modes in turn and they all have an unpleasant feature - after waking up ESP from deepsleep, a few lines of "red nonsense" are displayed. The ESP32C3 did not do this. So I don't know if the error is in Thonny's code or in Micropython's code.
![30](https://github.com/micropython/micropython/assets/81017132/72a369c7-f7bd-4985-9e60-8de4d4ee0a17)

**Reply by robert-hh** [reply]
This is an ESP8266 "feature". The first messages form the ROM bootloader are sent at a baudrate of 78440.

**Reply by DanSkareda** [reply]
Thanks for the explanation.
